keps/sig-api-machinery/1929-built-in-default/README.md

```diff
@@ -430,21 +437,27 @@ Object:
   properties:
     invalid:
       type: string
-      default: default-name
+      default: "default-name"
+	  # This field would only be present for CRDs, for built-in,
+	  # the empty value would be accepted and treated as missing.
+	  minLength: 1
     defaulted:
-      type: boolean
+      type: int
       default: false
 ```
 
 Which, for both CRD types and built-in types, would generate the following behavior:
 ```python
 >>> default('{}')
 {"entry": {"name": "default-name", "number": 0}}
->>> default('{entry: {name: other-name}}')
-{"entry": {"name": "other-name", "number": 0}}
+>>> default('{name: other-name}')
+{"name": "other-name", "number": 0}
+# For built-in, additionally, we would have:
+>>> default('{name: ""}')
+{"name": "default-name", "number": 0}
 ```
 
-Note that the generated code for built-in will look like this:
+Note that a simplified version of the generated code for built-ins would look like this:
```

Comment:
give an example of the planned generated code for a complex object as well

Reply:
That's not exactly what you asked, but I think it has the same purpose